02_oop_part_1/code/src/Sword.ts

```diff
@@ -0,0 +1,14 @@
+import { Weapon } from './Weapon';
+
+export class Sword extends Weapon {
+
+    constructor(baseDamage: number, baseDurability: number, value: number, weight: number) {
+        super('sword', baseDamage, baseDurability, value, weight)
+    }
+
+    polish() {
+        if(this.damageModifier < (this.baseDamage * 0.25)) {

```

Comment:
Where do you have this implementation -  If the base damage of a sword were to be 100, then the maximum that the effective damage could be increased to would be 125. ?

Reply:
this is the implementation, our damageModifier  cannot be more than 25% of base damage, and increasing is made on the next line... dont see the reason to make some new methods

Reply:
the question wasn’t about new methods. It was about the implementation itself, but actually, I got it the wrong way, so this implementation is enough.